src/XMakeCommandLine/XMake.cs

```diff
@@ -2835,10 +2835,7 @@ string unquotedParameter
             {
                 logger = loggerDescription.CreateLogger();
 
-                if (logger == null)
-                {
-                    InitializationException.VerifyThrow(logger != null, "LoggerNotFoundError", unquotedParameter);
-                }
+                InitializationException.VerifyThrow(logger != null, "LoggerNotFoundError", unquotedParameter);
```

Comment:
Oh, yeah, true, I should also look what's inside the if, not just at the condition :))